### PR TITLE
[TACHYON-1582] Restructure POM: cleanup  keyvalue module 

### PR DESCRIPTION
**Pull request by apc999**
https://tachyon.atlassian.net/browse/TACHYON-1582
This is the last PR to achieve what proposed in https://tachyon.atlassian.net/secure/attachment/14000/POMstructurechangeproposal.pdf

### CODE REVIEW

File: assembly/pom.xml [hunk 1]
```diff
@@ -21,12 +21,12 @@
...
     <dependency>
       <groupId>org.tachyonproject</groupId>
-      <artifactId>tachyon-core-common</artifactId>
+      <artifactId>tachyon-core-server</artifactId>
       <version>${project.version}</version>
     </dependency>
     <dependency>
       <groupId>org.tachyonproject</groupId>
-      <artifactId>tachyon-core-server</artifactId>
+      <artifactId>tachyon-keyvalue-server</artifactId>
```
Comment (apc999) [inline assembly/pom.xml]: now assembly jar contains keyvalue servers, and we enable/disable keyvalue feature by using `tachyon.keyvalue.enabled` property

File: keyvalue/client/pom.xml [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+  <parent>
+    <groupId>org.tachyonproject</groupId>
+    <artifactId>tachyon-keyvalue</artifactId>
+    <version>0.9.0-SNAPSHOT</version>
+  </parent>
+  <groupId>org.tachyonproject</groupId>
+  <artifactId>tachyon-keyvalue-client</artifactId>
+  <packaging>jar</packaging>
+  <name>Tachyon Key Value Store - Client Distribution</name>
+  <description>Distribution of client for Tachyon core</description>
```
Comment (uronce-cc) [inline keyvalue/client/pom.xml]: ... for Tachyon Key Value Store

**Comment by uronce-cc** [comment]
@apc999 LGTM, left one comment.

**Comment by ifcharming** [comment]
LGTM

**Comment by apc999** [comment]
Further consolidate module `tachyon-keyvalue-examples` with `tachyon-examples`, because (1) `tachyon-keyvalue-examples` depends on `tachyon-examples` which is less ideal and (2) reduce one module from probably a little bit too many modules.

**Comment by jsimsa** [comment]
LGTM

**Pull request merged**